presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java

```diff
@@ -1039,13 +1057,142 @@ public void testSelectWithGroupBy()
+                                selectList(new AllColumns()),
+                                Optional.of(new Table(QualifiedName.of("table1"))),
+                                Optional.empty(),
+                                Optional.empty(),
+                                Optional.empty(),
+                                ImmutableList.of(new WindowDefinition("a", new WindowSpecification(Optional.empty(), ImmutableList.of(new LongLiteral("1")), ImmutableList.of(), Optional.empty()))),
+                                ImmutableList.of(),
+                                Optional.empty()),
+                        ImmutableList.of(),
+                        Optional.empty(),
+                        Optional.empty()));
+
+        assertStatement("SELECT * FROM table1 WINDOW a AS (b PARTITION BY 1)",
+                new Query(
+                        Optional.empty(),
+                        new QuerySpecification(
+                                selectList(new AllColumns()),
+                                Optional.of(new Table(QualifiedName.of("table1"))),
+                                Optional.empty(),
+                                Optional.empty(),
+                                Optional.empty(),
+                                ImmutableList.of(new WindowDefinition("a", new WindowSpecification(Optional.of("b"), ImmutableList.of(new LongLiteral("1")), ImmutableList.of(), Optional.empty()))),
+                                ImmutableList.of(),
+                                Optional.empty()),
+                        ImmutableList.of(),
+                        Optional.empty(),
+                        Optional.empty()));
+
+        assertStatement("SELECT * FROM table1 WINDOW a AS (PARTITION BY 1), b AS (ORDER BY 2 ASC NULLS LAST)",
+                new Query(
+                        Optional.empty(),
+                        new QuerySpecification(
+                                selectList(new AllColumns()),
+                                Optional.of(new Table(QualifiedName.of("table1"))),
+                                Optional.empty(),
+                                Optional.empty(),
+                                Optional.empty(),
+                                ImmutableList.of(
+                                        new WindowDefinition("a", new WindowSpecification(Optional.empty(), ImmutableList.of(new LongLiteral("1")), ImmutableList.of(), Optional.empty())),
+                                        new WindowDefinition("b", new WindowSpecification(Optional.empty(), ImmutableList.of(), ImmutableList.of(new SortItem(new LongLiteral("2"), SortItem.Ordering.ASCENDING, SortItem.NullOrdering.LAST)), Optional.empty()))),
+                                ImmutableList.of(),
+                                Optional.empty()),
+                        ImmutableList.of(),
+                        Optional.empty(),
+                        Optional.empty()));
+    }
+
```

Comment:
Please add negative tests. E.g. invalid SQL failing, e.g. when trying to define window with the same name twice or using a name of window which is not defined.